src/network/clientpackethandler.cpp

```diff
@@ -982,6 +982,22 @@ void Client::handleCommand_SpawnParticle(NetworkPacket* pkt)
 	m_client_event_queue.push(event);
 }
 
+/* this function should not exist. alas, i managed to leave out critical fields
+ * in the original tween serialization/deserialization code and now the protocol
+ * is fixed, so now we all have to suffer for my abject stupidity. j o y */
+template <typename T>
+inline bool readMissingTweenParamsFor(std::istringstream& is, ParticleParamTypes::TweenedParameter<T>& t) {
```

Comment:
Read&write are TweenedParameter-specific functions, hence they should be put inside `TweenedParameter`.

Reply:
that is incorrect. they are helpers to make specific functions more readable, and should never be used from any other context. they will not be moved.